qiskit/circuit/quantumcircuit.py

```diff
@@ -818,14 +819,21 @@ def bind_parameters(self, value_dict):
         """
         new_circuit = self.copy()
 
-        if value_dict.keys() > self.parameters:
+        unrolled_value_dict = {}
+        for (param, value) in value_dict.items():
+            if isinstance(param, Parameter):
+                unrolled_value_dict[param] = value
+            if isinstance(param, ParameterVector):
+                unrolled_value_dict.update(zip(param, value))
```

Comment:
Can you add a check that `param` and `value` are the same length? (`zip` will stop the end of whichever is shorter)

Reply:
Done